viewing_party/party.py

```diff
@@ -1,23 +1,223 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict ={}
+    if title and genre and rating:
+        movie_dict["title"] = title 
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+        return movie_dict    
+    else:
+        return None
+
+
+def add_to_watched(user_data, movie):
+    watched_movies = user_data
+    watched_movies["watched"].append(movie)
+    return watched_movies
+
+
+def add_to_watchlist(user_data, movie):
+    watch_list = user_data
+    watch_list["watchlist"].append(movie)
+    return watch_list
+
+
+def watch_movie(user_data, movie_title):
+    update_watched = user_data
+    for movie in user_data["watchlist"]:
+        if movie_title in movie["title"]:
+            update_watched["watched"].append(movie)
+            update_watched["watchlist"].remove(movie)
+    print(update_watched)
+    return update_watched    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    
+    users_data = user_data
+    avg_values = 0
+    counter = 0
+    movies = users_data["watched"]
+    if movies:
+        for items in users_data.values():
+            for num in items:
+                avg_values += num["rating"]
+                counter += 1
+        result = avg_values/counter  
+        return result
+    else:
+        return 0
+
+# for items in users_data.values(): is'nt this the same as "movies"?
+# def get_most_watched_genre(user_data):
+#     users_data = user_data
+#     data = user_data.values()
+#     list = []   
+#     movies = users_data["watched"]
+
+#     if movies:
+#         for item in data:
+#             for i in item:
+#                 list.append(i["genre"])
+#         most_commom = max(set(list), key = list.count)
+#         return most_commom  
+#     else: 
+#         return most_commom
+
+
+def get_most_watched_genre(user_data):
+
+    watched_genres = {}
+    watched_movies = user_data["watched"]
+
+    if not watched_movies:
+        return None
+
+    for movie in watched_movies:
+        if movie["genre"] in watched_genres:
+            watched_genres[movie["genre"]] += 1
+        else: 
+            watched_genres[movie["genre"]] = 1
+    list_of_freq = list(watched_genres.values())
+    list_of_freq.sort()
+    most_watched_value = list_of_freq[-1]
+
+    for genre in watched_genres:
+        if watched_genres[genre] == most_watched_value:
+            most_watched_genre = genre
+    return most_watched_genre
+
+def friends_watched(user_data):

```

Comment:
I really like the idea of a helper function to build a unified list of the friends watched movies.

One thing that jumps out at me is that this seems to _not_ build the list if the user hasn't watched anything (the `if` on line 100)?

Also, we haven't talked about performance considerations yet, but it turns out that using `in` on a list actually hides a loop, so as the combined friends list gets longer, the length of time to see whether the next movie is already in the list also takes longer. We could use a helper set of movie titles to help us do this check more efficiently. Each time we add a movie to the combined list, we could add its title to a set. Then, when deciding whether or not to add the next movie, we could check whether that movie's title is in the title set already. `in` on sets is much more efficient than on lists.